packages/mcp-servers/devtool-mcp-server/src/tools/Tracing/EndTracing.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+import { clientId } from '../../schema/index.ts';
+import { defineTool } from '../defineTool.ts';
+import type { TracingComplete } from '../../connector.ts';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import * as os from 'node:os';
+import { Writable } from 'node:stream';
+
+type IOReadResponse = {
+  data: string;
+  eof: boolean;
+};
+
+export const EndTracing = /*#__PURE__*/ defineTool({
+  name: 'Tracing_end',
+  description: 'Stop trace events collection',
+  schema: {
+    clientId,
+  },
+  annotations: {
+    readOnlyHint: false,
+  },
+  async handler({ params }, response, context) {
+    const connector = context.connector();
+    const sendIOReadMessage = (stream: number): Promise<IOReadResponse> => {
+      return connector.sendCDPMessage<IOReadResponse>(
+        params.clientId,
+        -1,
+        'IO.read',
```

Comment:
Do we also need to close the IO stream? E.g.: sending `IO.close` CDP request. See: https://chromedevtools.github.io/devtools-protocol/tot/IO/#method-close